### PR TITLE
[vcpkg toolchain] Add help info to help users find out why port find failed

### LINKED ISSUE #30247: [vcpkg toolchain] Add help information after find port failed
We have received multiple feedbacks about the failure of using `find_package`, maybe we can fuzzy find the required _port:triplet_ after the cmake configuration fails to return help information.

### PR DESCRIPTION
**Pull request by JackBoosY**
After finding the port fails when the user uses the vcpkg toolchain:
1. Print a similar installed message when the port exists: check the installed port and its triplet through the `list` command.
2. Print cmake default information when the port is not included in vcpkg.

Fixes https://github.com/microsoft/vcpkg/issues/30247

### CODE REVIEW

File: scripts/buildsystems/vcpkg.cmake [hunk 1]
```diff
@@ -845,6 +847,47 @@ macro("${VCPKG_OVERRIDE_FIND_PACKAGE_NAME}" z_vcpkg_find_package_package_name)
...
+                ${Z_VCPKG_ADDITIONAL_MANIFEST_PARAMS}
+                ${VCPKG_INSTALL_OPTIONS}
+            OUTPUT_VARIABLE maybe_installed_out
+            RESULT_VARIABLE maybe_installed_res
+        )
+        
+        set(print_vcpkg_help OFF)
+        string(TOLOWER "${z_vcpkg_find_package_package_name}" z_vcpkg_find_package_package_name_lower)
+        if (NOT maybe_installed_res EQUAL 0)
+            string(REGEX MATCHALL "^([^ ]+)" maybe_installed_port "${maybe_installed_out}")
```
Comment (JackBoosY) [inline scripts/buildsystems/vcpkg.cmake:867]: We should also consider the port is not installed in any triplets.

File: scripts/buildsystems/vcpkg.cmake [hunk 2]
```diff
@@ -845,6 +847,47 @@ macro("${VCPKG_OVERRIDE_FIND_PACKAGE_NAME}" z_vcpkg_find_package_package_name)
     else()
         _find_package("${z_vcpkg_find_package_package_name}" ${z_vcpkg_find_package_${z_vcpkg_find_package_backup_id}_ARGN})
     endif()
+
+    if (NOT ${z_vcpkg_find_package_package_name}_FOUND)
```
Comment (Neumann-A) [inline scripts/buildsystems/vcpkg.cmake:851]: That is not enough to be correct. `${z_vcpkg_find_package_package_name}_FOUND` can be any case and there is even a cmake warning if the variable doesn't agree with the call. Furthermore this only needs to be shown if the find_package call was REQUIRED.

File: scripts/buildsystems/vcpkg.cmake (hunk 1 above)
Comment (Neumann-A) [reply scripts/buildsystems/vcpkg.cmake:867]: you should also consider `port name` != `package name`. 
Basically you cannot do this without a file mapping ports to possible cmake packages.

File: scripts/buildsystems/vcpkg.cmake [hunk 3]
```diff
@@ -845,6 +847,47 @@ macro("${VCPKG_OVERRIDE_FIND_PACKAGE_NAME}" z_vcpkg_find_package_package_name)
...
+                set(print_vcpkg_help ON)
+            endif()
+        endif()
+
+        if (NOT print_vcpkg_help)
+            message(FATAL_ERROR "Could not find a package configuration file provided by \"${z_vcpkg_find_package_package_name}\" "
+                "with any of the follwing names:\n"
+                "${z_vcpkg_find_package_package_name}Config.cmake\n"
+                "${z_vcpkg_find_package_package_name_lower}-config.cmake\n"
+                "This package is not provided by vcpkg.\n"
```
Comment (Neumann-A) [inline scripts/buildsystems/vcpkg.cmake:883]: This conclusion seems to be wrong.

File: scripts/buildsystems/vcpkg.cmake (hunk 1 above)
Comment (JackBoosY) [reply scripts/buildsystems/vcpkg.cmake:867]: Yeah, we can't get the corresponding port name on the cmake side (the _packages_ path may not exist so it cannot be used), and cmake finds it by looking for the cmake file in the _share_ path.
So the best way is that the vcpkg binary provides this interface to get this dependency through the vcpkg database.

**Comment by dg0yt** [comment]
I don't think it is possible to reliably determine why `find_package` failed. There are too many possible reasons:
- package name case mismatch
- version mismatch
- missing components
- failed dependency
- whatever else the extended siganture of `find_package` supports
- whatever else could be poorly implemented in wrappers or in find modules.

CMake already has a message which tries to explain the reasons. `VCPKG_TRACE_FIND_PACKAGE` offers additional information, and there is also `CMAKE_FIND_DEBUG_MODE`.

However, the tool might faily reliable print a negative message:
~~~
<port> does not provide CMake config.
~~~
(However, `usage` would be needed for all ports which are supported by official Find modules or which can be used via pkg-config.)

**Pull request closed without merging**